tests/hw18.spec.js

```diff
@@ -0,0 +1,169 @@
+        await expect(page.getByText('Last name has to be from 2 to 20 characters long')).toBeHidden();
+        await page.locator('#signupLastName').fill('test123test123test12');
+        await expect(page.getByText('Last name has to be from 2 to 20 characters long')).toBeHidden();
+        await page.locator('#signupLastName').fill('test123test123test123');
+        await expect(page.getByText('Last name has to be from 2 to 20 characters long')).toBeVisible();
+    })
+
+    test('"Email" input field validation', async ({page})=>{
+        // Empty Email field validation
+        await page.locator('#signupEmail').focus();
+        await page.locator('#signupEmail').blur();
+
+        // Border of input field is red
+        await expect(page.locator('#signupEmail')).toHaveCSS('border-color', 'rgb(220, 53, 69)');
+
+        await expect(page.getByText('Email required')).toBeVisible();
+        await page.locator('#signupEmail').fill('test@test.test');
+        await expect(page.getByText('Email required')).toBeHidden();
+
+        // Incorrect Email validation
+        await page.locator('#signupEmail').fill('test');
+        await expect(page.getByText('Email is incorrect')).toBeVisible();
+        await page.locator('#signupEmail').fill('test@');
+        await expect(page.getByText('Email is incorrect')).toBeVisible();
+        await page.locator('#signupEmail').fill('test@test.');
+        await expect(page.getByText('Email is incorrect')).toBeVisible();
+        await page.locator('#signupEmail').fill('test@test.test');
+        await expect(page.getByText('Email is incorrect')).toBeHidden();
+    })
+    
+    test('"Password" input field validation', async ({page})=>{
+        const passwordInput = page.getByLabel('Password', { exact: true });
+        const invalidPWvalidationError = 'Password has to be from 8 to 15 characters long and contain at least one integer, one capital, and one small letter';
+        // Empty Password field validation
+        await passwordInput.focus();
+        await passwordInput.blur();
+
+        // Border of input field is red
+        await expect(passwordInput).toHaveCSS('border-color', 'rgb(220, 53, 69)');
+
+        await expect(page.getByText('Password required')).toBeVisible();
+
+        // Invalid Password field validation
+        await passwordInput.fill('qwertyu');
+        await expect(page.getByText(`${invalidPWvalidationError}`)).toBeVisible();
+        await passwordInput.fill('Qwerty1');
+        await expect(page.getByText(`${invalidPWvalidationError}`)).toBeVisible();
+        await passwordInput.fill('Qwertyui');
+        await expect(page.getByText(`${invalidPWvalidationError}`)).toBeVisible();
+        await passwordInput.fill('QwertyuiQwertyu1');
+        await expect(page.getByText(`${invalidPWvalidationError}`)).toBeVisible();
+        await passwordInput.fill('Qwertyui1');
+        await expect(page.getByText(`${invalidPWvalidationError}`)).toBeHidden();
```

Comment:
how do you know this test will appear in registration popup but not on other parts of page ?